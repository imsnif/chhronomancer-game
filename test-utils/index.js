module.exports = {
  mockFB () {
    const login = require('react-native-facebook-login')
    const FBLogin = jest.fn()
    FBLogin.mockReturnValue(null)
    login.FBLogin = FBLogin
    login.FBLoginManager = {LoginBehaviors: {Web: 2}}
    return FBLogin
  },
  navigatorMock (routeName) {
    return {
      getCurrentRoutes: () => [routeName],
      push: () => {}
    }
  }
}
