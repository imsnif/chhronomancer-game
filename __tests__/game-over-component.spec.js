import React from 'react'
import 'react-native'
import renderer from 'react-test-renderer'
import serverMock from '../server-mock'
import { navigatorMock } from '../test-utils'
import statsStore from '../stores/stats'

jest.mock('mobx-react/native', () => require('mobx-react/custom'))

beforeEach(() => {
  jest.useFakeTimers()
  serverMock()
})

afterEach(() => {
  const clockStore = require('../stores/clock').default
  jest.useRealTimers()
  clockStore.destroy()
  jest.useFakeTimers()
  statsStore.clearWinner()
})

test('GameOver (with winner) => renders correctly', () => {
  statsStore.announceWinner(1)
  const GameOver = require('../components/game-over').default
  const tree = renderer.create(
    <GameOver navigator={navigatorMock({screenName: 'foo'})} />
  ).toJSON()
  expect(tree).toMatchSnapshot()
})

test('GameOver (no winner) => renders empty', () => {
  statsStore.clearWinner()
  const GameOver = require('../components/game-over').default
  const tree = renderer.create(
    <GameOver navigator={navigatorMock({screenName: 'foo'})} />
  ).toJSON()
  expect(tree).toMatchSnapshot()
})
