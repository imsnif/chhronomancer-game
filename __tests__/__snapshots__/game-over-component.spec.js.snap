// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GameOver (no winner) => renders empty 1`] = `null`;

exports[`GameOver (with winner) => renders correctly 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#141313",
      "flexDirection": "column",
      "height": "100%",
      "paddingBottom": 5,
      "paddingTop": 5,
      "width": "100%",
    }
  }
>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Object {
        "color": "#329932",
        "fontSize": 45.911331934501746,
        "height": 40.019999999999996,
        "includeFontPadding": false,
        "lineHeight": 15,
        "paddingLeft": 15,
        "paddingRight": 15,
        "textAlignVertical": "top",
      }
    }
  >
    +------------------------------------------------------+
  </Text>
  <View
    style={
      Object {
        "flex": 1,
        "flexDirection": "row",
      }
    }
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "color": "#329932",
          "flex": 0,
          "lineHeight": 13,
          "textAlignVertical": "top",
          "width": 22.5,
        }
      }
    >
       | 
 | 
 | 
 | 
 | 
 | 
 | 
 | 
 | 
 | 
 | 
 | 
 | 
 | 
 | 
 | 
 | 
 | 
 | 
 | 
 | 
 | 
 | 
 | 
 | 
 | 
 | 
 | 
 | 
 | 
 | 
 | 
 | 
 | 
 | 
 | 
 | 
 | 
 | 
 | 
 | 

    </Text>
    <View
      style={
        Object {
          "backgroundColor": "#141313",
          "flexDirection": "column",
          "height": "100%",
          "maxWidth": 291,
          "paddingBottom": 5,
          "paddingTop": 5,
          "width": "100%",
        }
      }
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "alignSelf": "center",
            "backgroundColor": "#141313",
            "color": "#329932",
            "flex": 1,
            "fontFamily": "telegrama_raw",
            "fontSize": 25,
            "textAlign": "center",
            "textAlignVertical": "center",
          }
        }
      >
        GAME OVER
      </Text>
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "alignSelf": "center",
            "backgroundColor": "#141313",
            "color": "#329932",
            "flex": 1,
            "fontFamily": "telegrama_raw",
            "fontSize": 25,
            "textAlign": "center",
            "textAlignVertical": "center",
          }
        }
      >
        Winner: 
        Aram
      </Text>
      <Image
        source={
          Object {
            "uri": "http://10.0.0.6:3000/placeholders/nopic.png",
          }
        }
        style={
          Object {
            "alignSelf": "center",
            "backgroundColor": "#141313",
            "borderColor": "black",
            "flex": 2,
            "marginBottom": 20,
            "maxHeight": 200,
            "maxWidth": 200,
            "resizeMode": "stretch",
          }
        }
      >
        <View
          style={
            Object {
              "backgroundColor": "green",
              "height": "100%",
              "opacity": 0.5,
              "width": "100%",
            }
          }
        />
      </Image>
    </View>
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "color": "#329932",
          "flex": 0,
          "lineHeight": 13,
          "textAlignVertical": "top",
          "width": 22.5,
        }
      }
    >
       | 
 | 
 | 
 | 
 | 
 | 
 | 
 | 
 | 
 | 
 | 
 | 
 | 
 | 
 | 
 | 
 | 
 | 
 | 
 | 
 | 
 | 
 | 
 | 
 | 
 | 
 | 
 | 
 | 
 | 
 | 
 | 
 | 
 | 
 | 
 | 
 | 
 | 
 | 
 | 
 | 

    </Text>
  </View>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Object {
        "color": "#329932",
        "fontSize": 45.911331934501746,
        "height": 40.019999999999996,
        "includeFontPadding": false,
        "lineHeight": 15,
        "paddingLeft": 15,
        "paddingRight": 15,
        "textAlignVertical": "top",
      }
    }
  >
    +------------------------------------------------------+
  </Text>
</View>
`;
